databricks/koalas/typedef/typehints.py

```diff
@@ -93,41 +93,48 @@ class NameTypeHolder(object):
 
 def as_spark_type(tpe) -> types.DataType:
     """
-    Given a python type, returns the equivalent spark type.
+    Given a Python type, returns the equivalent spark type.
     Accepts:
-    - the built-in types in python
+    - the built-in types in Python
     - the built-in types in numpy
     - list of pairs of (field_name, type)
     - dictionaries of field_name -> type
-    - python3's typing system
+    - Python3's typing system
     """
-    if tpe in (str, "str", "string"):
-        return types.StringType()
+    # ArrayType
+    if tpe in (np.ndarray,):
+        # TODO: support other child types
+        return types.ArrayType(types.StringType())
+    # BinaryType
     elif tpe in (bytes,):
         return types.BinaryType()
+    # BooleanType
+    elif tpe in (bool, np.bool, "bool", "boolean"):
+        return types.BooleanType()
+    # DateType
+    elif tpe in (datetime.date,):
+        return types.DateType()
+    # NumericType
     elif tpe in (np.int8, "int8", "byte"):
         return types.ByteType()
-    elif tpe in (np.int16, "int16", "short"):
-        return types.ShortType()
-    elif tpe in (int, "int", np.int, np.int32):
-        return types.IntegerType()
-    elif tpe in (np.int64, "int64", "long", "bigint"):
-        return types.LongType()
-    elif tpe in (float, "float", np.float):
-        return types.FloatType()
-    elif tpe in (np.float64, "float64", "double"):
-        return types.DoubleType()
     elif tpe in (decimal.Decimal,):
         return types.DecimalType(38, 18)
-    elif tpe in (datetime.datetime, np.datetime64):
+    elif tpe in (float, np.float, np.float64, "float", "float64", "double"):
+        return types.DoubleType()
+    elif tpe in (np.float32, "float32"):
+        return types.FloatType()
+    elif tpe in (np.int32, "int32"):
+        return types.IntegerType()
+    elif tpe in (int, np.int, np.int64, "int", "int64", "long", "bigint"):
+        return types.LongType()
+    elif tpe in (np.int16, "int16", "short"):
+        return types.ShortType()
+    # StringType
+    elif tpe in (str, "str", "string"):
+        return types.StringType()
+    # TimestampType
+    elif tpe in (datetime.datetime, np.datetime64, "datetime64[ns]"):
```

Comment:
also `pd.Timestamp`?

Reply:
I think maybe then we have inconsistency behavior with pandas for `astype` ??

```python
>>> pd.Series(["2020-10-20"]).astype(pd.Timestamp)
Traceback (most recent call last):
...
TypeError: dtype '<class 'pandas._libs.tslibs.timestamps.Timestamp'>' not understood

>>> ks.Series(["2020-10-20"]).astype(pd.Timestamp)
0   2020-10-20
dtype: datetime64[ns]
```

What do you think about this ??

Reply:
Got it. Let's not add it for now. We might need to add it later anyway since the function is used not only in `astype`.